src/Common/src/Common/Reflection/ReflectionHelpers.cs

```diff
@@ -304,30 +305,37 @@ private static void TryLoadAssembliesWithAttribute<T>()
         where T : AssemblyContainsTypeAttribute
     {
         var runtimeAssemblies = Directory.GetFiles(RuntimeEnvironment.GetRuntimeDirectory(), "*.dll");
-        using var loadContext = new MetadataLoadContext(new PathAssemblyResolver(AllRelevantPaths(runtimeAssemblies, typeof(T))));
-        var loadedAssemblies = AppDomain.CurrentDomain.GetAssemblies();
-        var assemblypaths = Directory.EnumerateFiles(AppDomain.CurrentDomain.BaseDirectory).Where(f => f.EndsWith("dll", StringComparison.InvariantCultureIgnoreCase));
-        foreach (var assembly in assemblypaths)
+        try
         {
-            var filename = Path.GetFileNameWithoutExtension(assembly);
-            if (!loadedAssemblies.Any(a => a.FullName.StartsWith(filename, StringComparison.InvariantCultureIgnoreCase)))
+            using var loadContext = new MetadataLoadContext(new PathAssemblyResolver(AllRelevantPaths(runtimeAssemblies, typeof(T))));
```

Comment:
Working off an empty list rather than using try/catch:
```csharp
        var allKnownAssemblyPaths = AllRelevantPaths(runtimeAssemblies, typeof(T));
        if (!allKnownAssemblyPaths.Any())
        {
            return;
        }

        using var loadContext = new MetadataLoadContext(new PathAssemblyResolver(allKnownAssemblyPaths));
```

Reply:
Change has been made as suggested.